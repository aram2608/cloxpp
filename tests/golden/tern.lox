// Basic conditional
print 1 < 2 ? "yes" : "no"; // expect "yes"

// False branch
print 2 < 1 ? "oops" : "works"; // expect "works"

// Nesting
print true ? (false ? "inner-yes" : "inner-no") : "outer-no"; 
// expect "inner-no"

// With variables
var x = 10;
var y = 20;
print x > y ? x : y; // expect 20

// Inside an expression
print (true ? 5 : 100) + 3; // expect 8

// Mixing with logical ops
print false or (true ? true : false); // expect true

// Edge: chained conditionals
print 5 == 5 ? "equal" : 5 < 5 ? "less" : "greater";
// expect "equal"
