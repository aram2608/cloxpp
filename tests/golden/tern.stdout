yes
works
inner-no
20.000000
8.000000
true
equal
